### DISCUSSION TITLE
SignalR

### DISCUSSION THREAD

**Discussion by JoshuaNitschke** [hydrostack/hydro#44] [Ideas]
Is it on your radar to integrate with Signal R? I was considering using Angular for my side project but after stumbling across this repo on reddit, I decided to give it a go as I want to save time remodeling my domain in Angular and in general I am more comfortable in dotnet than typescript, subscriptions, etc.  However, I had planned on using signal r for reeltime event pub/sub as my app as a decent amount of real time activity.  I see we have a polling option and I could make that work, but just curious if we could have a Signal R or some kind of sockets based functionality as an add on ?

**Comment by kjeske** [comment]
Hi! Yes, I thought about SignalR, but the way how web sockets work and how they are handled introduces quite some challenges. Hydro operates on regular HTTP request/response pipeline with all the Razor tools that base on that. To make it work with pure web socket connection would be probably a big project. I'm not saying no, but I don't think it will be in Hydro soon.

What about [Service-sent events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)? It's a one direction communication from the server to the client. That would be easier to implement in Hydro and I would see that soon. Then the client to server communication can be done using regular Hydro actions. Would that work?

Also thank you for supporting me on Github Sponsors!

**Reply by JoshuaNitschke** [reply]
Sorry, I missed your original reply.  Yes, this sounds like it could work well.

**Comment by kjeske** [comment]
Closing for now since no new messages, but please open in case of new questions/thoughts. Server-sent events will be added to Hydro in the next releases.
